src/Writer/PhpArray.php

```diff
@@ -139,4 +158,54 @@ protected function processIndented(array $config, array $arraySyntax, &$indentLe
 
         return $arrayString;
     }
+
+    /**
+     * Process a string configuration value
+     *
+     * @param string $value
+     * @return string
+     */
+    protected function processStringValue($value)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($value)) {
+            return $value . '::class';
+        }
+
+        return var_export($value, true);
+    }
+
+    /**
+     * Process a string configuration key
+     *
+     * @param string $key
+     * @return string
+     */
+    protected function processStringKey($key)
+    {
+        if ($this->useClassNameScalars && $this->checkStringIsFqn($key)) {
+            return $key . '::class';
+        }
+
+        return "'" . addslashes($key) . "'";
+    }
+
+    /**
+     * Check whether a string represents a resolvable FQCN
+     *
+     * @param string $string
+     * @return bool
+     */
+    protected function checkStringIsFqn($string)
```

Comment:
That is what I thought, but ocramius noticed the following issue correctly: https://3v4l.org/ApMIA
Say you make an alias for a class named `*`, class_exists would return `true` in config keys and values containing `*` character: `*::class`.

This would happen with other aliased string sequences too. This is just an example.